common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -461,7 +463,7 @@ public static String bytesToIpAddress(byte[] bytes, int offset, int length) {
         }
     }
 
-    public static boolean isValidIpV6Address(String ip) {
+    public static boolean isValidIpV6Address(CharSequence ip) {
```

Comment:
We also need to keep the old method as otherwise the change is not binary compatible.

Reply:
good catch ... done